src/test/java/org/folio/rest/impl/PatronResourceImplTest.java

```diff
@@ -117,6 +118,7 @@
   private final String intransitItemId = "32e5757d-6566-466e-b69d-994eb33d2c98";
   private static final String availableItemId = "32e5757d-6566-466e-b69d-994eb33d2b62";
   private static final String checkedoutItemId = "32e5757d-6566-466e-b69d-994eb33d2b73";
+  private static final String sortByParam = "item.title/sort.ascending";
```

Comment:
How does this example work in practice, given that `item.title` is not a path that exists for [loans](https://github.com/folio-org/mod-circulation-storage/blob/master/ramls/loan.json) or [accounts](https://github.com/folio-org/mod-feesfines/blob/master/ramls/accountdata.json)?

Reply:
@marcjohnson-kint I think the expectation is that if you're specifying sortBy, you'll only be using one of: [ includeHolds=true, includeLoans=true, includeCharges=true ].

From the Story description:  
> If the sort field doesn't exist all three tables, that's fine.  The results for that list just won't be sorted.  The assumption is that in those cases multiple requests would be made to the API to get Holds (sorted by X), Loans (sorted by Y), and Charges (sorted by Z).

Reply:
@craigmcnally  I think I don't understand why this work is needed. It might be best I don't review the changes.

> If the sort field doesn't exist all three tables, that's fine. The results for that list just won't be sorted. The assumption is that in those cases multiple requests would be made to the API to get Holds (sorted by X), Loans (sorted by Y), and Charges (sorted by Z).

This seems like a really strange trade off to me, that we want clients to make multiple requests to an API (increasing load on the system) solely in order to sort some sub-elements of a response differently.

Are the lists of holds, loans and charges intended to be complete in the response from the API?

If so, why do we need the server to do the sorting rather than the client?

Reply:
@marcjohnson-kint these APIs are being consumed by several clients which share a common need - to display a patron's loans, holds, and fees in three separate tabs/sections, each independent of the others.  A patron might want to view their holds but never click to view their loans or fees.  Fetching the three independently seems reasonable in these cases.  Furthermore, the sort dimensions for the three lists will be different too... a fee/charge for instance may not have a title (locker rental fees, or photocopy charges, etc.)  Another example is that we might want to sort by due date, which might only apply to loans.  So sorting all three lists by the same set of fields isn't necessarily applicable.

I'll point out that clients can always choose to **not** provide sort criteria and do the sorting on the client side if they wish.

Reply:
@craigmcnally I get the strong sense I don't understand the use cases or design for this API.

I think I should stop reviewing this change.

> these APIs are being consumed by several clients which share a common need - to display a patron's loans, holds, and fees in three separate tabs/sections, each independent of the others. A patron might want to view their holds but never click to view their loans or fees. Fetching the three independently seems reasonable in these cases.

That suggests to me that these needs could be better met with separate APIs.

>  Furthermore, the sort dimensions for the three lists will be different too... a fee/charge for instance may not have a title (locker rental fees, or photocopy charges, etc.) Another example is that we might want to sort by due date, which might only apply to loans. So sorting all three lists by the same set of fields isn't necessarily applicable.

That suggests to me that these needs could be better met with separate sort criteria.